airflow/ti_deps/ti_deps.py

```diff
@@ -0,0 +1,479 @@
+from abc import ABCMeta, abstractmethod
```

Comment:
I'm not sure why, but you are right it's not very commonly used. I still think it has some fairly substantial benefits but will remove it.